client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/LicensingDocumentationIT.java

```diff
@@ -167,4 +170,63 @@ public void onFailure(Exception e) {
             assertThat(currentLicense, endsWith("}"));
         }
     }
+
+    public void testPostStartTrial() throws Exception {
+        RestHighLevelClient client = highLevelClient();
+        {
+            // tag::post-start-trial-execute
+            PostStartTrialRequest request = new PostStartTrialRequest();
+            request.acknowledge(false);
+
+            PostStartTrialResponse response = client.license().postStartTrial(request, RequestOptions.DEFAULT);
+            // end::post-start-trial-execute
+
+            // tag::post-start-trial-response
+            boolean acknowledged = response.isAcknowledged();
+            boolean trialWasStarted = response.isTrialWasStarted();
+            String errorMessage = response.getErrorMessage();
+            String type = response.getType();
+            String acknowledgeMessage = response.getAcknowledgeMessage(); // todo rename this to acknowledgeheader for consistency
+            Map<String, String[]> acknowledgeMessages = response.getAcknowledgeMessages();
+            // end::post-start-trial-response
+
+            assertFalse(acknowledged);
+            assertFalse(trialWasStarted);
+            assertEquals("Operation failed: Needs acknowledgement.", errorMessage);
+            assertNull(type);
+            assertThat(acknowledgeMessage, containsString("To begin your free trial, call /start_trial again and specify " +
+                "the \"acknowledge=true\" parameter."));
+            assertThat(acknowledgeMessages.entrySet(), not(empty()));
+        }
+
+        {
+            PostStartTrialRequest request = new PostStartTrialRequest();
+
+            // tag::post-start-trial-execute-listener
+            ActionListener<PostStartTrialResponse> listener = new ActionListener<PostStartTrialResponse>() {
+                @Override
+                public void onResponse(PostStartTrialResponse postStartTrialResponse) {
+                    // <1>
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    // <2>
+                }
+            };
+            // end::post-start-trial-execute-listener
+
+            final CountDownLatch latch = new CountDownLatch(1);
+            listener = new LatchedActionListener<>(listener, latch);
+
+            // tag::post-start-trial-execute-async
+            client.license().postStartTrialAsync(request, RequestOptions.DEFAULT, listener);
+            // end::post-start-trial-execute-async
+
+            assertTrue(latch.await(30L, TimeUnit.SECONDS));
+
+            // todo add some other cases with randomization
+            // todo add a case that succeeds in starting the trial
```

Comment:
I'm not sure there is an IT test case for this anywhere, although I need to look more thoroughly